src/cmd/test.go

```diff
@@ -32,16 +34,21 @@ var testCmd = &cobra.Command{
 			buildMode.Mount = false
 		}
 
-		var testsPassed bool
+		// Listen to SIGINT
+		shutdownChannel := make(chan os.Signal, 1)
+		signal.Notify(shutdownChannel, os.Interrupt)
+
+		var testResult types.TestResult
 		if context.HasServiceContext {
-			testsPassed, err = application.TestService(context.ServiceContext, writer, buildMode)
+			testResult, err = application.TestService(context.ServiceContext, writer, buildMode, shutdownChannel)
 		} else {
-			testsPassed, err = application.TestApp(context.AppContext, writer, buildMode)
+			testResult, err = application.TestApp(context.AppContext, writer, buildMode, shutdownChannel)
 		}
 		if err != nil {
 			panic(err)
 		}
-		if !testsPassed {
+		signal.Stop(shutdownChannel)
+		if !testResult.Passed && !testResult.Interrupted {
```

Comment:
I think we can go leave this as it was: `if !testsPassed`. passed should be false on interrupt and the exit code also doesn't really matter during interrupt.